src/deploy/functions/cache/hash.ts

```diff
@@ -2,52 +2,37 @@ import { readFile } from "node:fs/promises";
 import * as crypto from "crypto";
 import { Backend, Endpoint } from "../backend";
 import { getSecretVersions } from "../../../functions/secrets";
+import { BinaryLike } from "crypto";
 
 /**
  * Generates a hash from the environment variables of a {@link Backend}.
  * @param backend Backend of a set of functions
  */
 export function getEnvironmentVariablesHash(backend: Backend): string {
-  const hash = crypto.createHash("sha256");
-
   // Hash the contents of the dotenv variables
   const hasEnvironmentVariables = !!Object.keys(backend.environmentVariables).length;
-  if (hasEnvironmentVariables) {
-    hash.update(JSON.stringify(backend.environmentVariables));
-  }
-
-  return hash.digest("hex");
+  return createHash(hasEnvironmentVariables ? JSON.stringify(backend.environmentVariables) : "");
 }
 
 /**
- * Retrieves the unique hash given a pathToGeneratedPackageFile.
- * @param pathToGeneratedPackageFile Packaged file contents of functions
+ * Retrieves the unique hash given a pathToFile.
+ * @param pathToFile Packaged file contents of functions
  */
-export async function getSourceHash(pathToGeneratedPackageFile: string): Promise<string> {
-  const hash = crypto.createHash("sha256");
-
-  // Hash the contents of the source file
-  const data = await readFile(pathToGeneratedPackageFile);
-  hash.update(data);
-
-  return hash.digest("hex");
+export async function getSourceHash(pathToFile: string): Promise<string> {
+  // Hash the contents of a file
+  const data = await readFile(pathToFile);
+  return createHash(data);
 }
 
 /**
  * Retrieves a hash generated from the secrets of an {@link Endpoint}.
  * @param endpoint Endpoint
  */
 export function getSecretsHash(endpoint: Endpoint): string {
-  const hash = crypto.createHash("sha256");
-
   // Hash the secret versions.
   const secretVersions = getSecretVersions(endpoint);
   const hasSecretVersions = !!Object.keys(secretVersions).length;
-  if (hasSecretVersions) {
-    hash.update(JSON.stringify(secretVersions));
-  }
-
-  return hash.digest("hex");
+  return createHash(hasSecretVersions ? JSON.stringify(secretVersions) : "");
```

Comment:
similar comment to removing `hasEnvironmentVariable`.

Reply:
Updated!